packages/repository/src/__tests__/integration/repositories/resolve-has-many-metadata.integration.ts

```diff
@@ -0,0 +1,164 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/repository
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect} from '@loopback/testlab';
+import {
+  Entity,
+  HasManyDefinition,
+  ModelDefinition,
+  RelationType,
+} from '../../..';
+import {resolveHasManyMetadata} from '../../../relations/has-many/has-many.helpers';
+
+describe('keyTo and keyFrom with resolveHasManyMetadata', () => {
+  it('resolves metadata using keyTo and keyFrom', () => {
+    const meta = resolveHasManyMetadata(Category.definition.relations[
+      'products'
+    ] as HasManyDefinition);
+
+    expect(meta).to.eql({
+      name: 'products',
+      type: 'hasMany',
+      targetsMany: true,
+      source: Category,
+      keyFrom: 'id',
+      target: () => Product,
+      keyTo: 'categoryId',
+    });
+  });
+
+  it('resolves metadata using keyTo, but not keyFrom', () => {
+    const meta = resolveHasManyMetadata(Category.definition.relations[
+      'items'
+    ] as HasManyDefinition);
+
+    expect(meta).to.not.have.property('keyFrom');
```

Comment:
If `keyFrom` is not a "must have" property, shall we define it as optional in the type `hasOneResolvedDefinition`?
The corresponding code: https://github.com/strongloop/loopback-next/pull/3726/files#diff-cbf88c051be861b9a242238ab4c6576cR19

Reply:
If it doesn't have `keyFrom`, `keyFrom` won't be included in the metadata. Do we need to note it as optional property in this case?

Reply:
I think it's fine as it is now. Like @agnes512 said, if we don't have it then it just won't include it in the metadata.

Reply:
@jannyHou I realized I made a mistake before, but now when `keyFrom` isn't included, then it gets [inferred](https://github.com/strongloop/loopback-next/pull/3726/files#diff-cbf88c051be861b9a242238ab4c6576cR48) based on the source model's `id` property.